0006-sync-iteration/README.md

```diff
@@ -0,0 +1,197 @@
+- individual records limited to 1 MByte
+- maximum of 200 record operations per commit
+
+Note that limits can not be maximized together: you can't have a commit with 10 records, each at the maximum size. Users and application designers should steer well clear of all these limits. For example, records should be less than 10 KByte in almost all situations; blobs should be used for larger data.
+
+
+## Partial Synchronization
+
+As the atproto network has grown, it has become more costly and inefficient to process every single account or every single record in the network. This is particularly true when working with applications using novel Lexicon schemas: in that situation, most of the 30 million repositories in the network contain no relevant records; and for the repositories which do, only a small fraction of the records in the repository are relevant. As part of the overall Sync 1.1 project, we are adding features to help with both of these challenges.
+
+To help with identifying the subset of repositories (DIDs) which are relevant to a given application, there is a new `com.atproto.sync.listReposByCollection` API endpoint. This takes a collection (NSID), and enumerates a list of all DIDs in the network which contain records of that type (using a cursor for pagination). This is particularly helpful when bootstrapping services from scratch. This endpoint may be implemented by relays, though it should be considered an optional additional service, not a required part of the relay role in the network.
+
+To help with fetching a subset of records from a repository, a variant of the `com.atproto.sync.getRepo` endpoint will be specified which allows fetching a "subtree" which includes records matching a provided collection prefix (which might include multiple collections/NSIDs). The commit block and all records in the range would be included, as well as MST nodes "proving" the records, as well as "proving" the keys (if any) directly adjacent to the key range, which provides a "proof" that there are no other matching keys.
+
+
+## Streaming CAR Processing
+
+Implementing automatic re-sychronization will involve more `com.atproto.sync.getRepo` fetches by consuming services. It is important to make these efficient to process.
+
+One mechanism is to request only the record range needed, as described above in "Partial Synchronization".
+
+Another is to make it possible to read repo CAR files in a "streaming" fashion, instead of loading the full tree into memory. The common case is to "walk" the keys in the MST data structure, starting from the commit block and then descending in depth-first order starting with the lowest key. If the blocks in the CAR file are in just the right order, then the tree can be walked by reading a single block at a time, and retaining only a few "parent" parsed tree nodes in memory.
+
+The exact order requirement will need to be formalized more, but would be roughly:
```

Comment:
I do think that "commit block then preorder traversal of the MST" is pretty darn close to the formal definition we're looking for and is descriptive enough for readers to get the gist, help "see" how it permits streaming verification.

Reply:
tweaked terms and reduced hedging